neat_cache/lib/src/providers/redis.dart

```diff
@@ -54,19 +53,22 @@ class RedisCacheProvider extends CacheProvider<List<int>> {
 
   Future<_RedisContext> _createContext() async {
     try {
-      _logger.info('Connecting to redis');
-      final connection = await Connection.connect(_connectionString)
-          .timeout(_connectTimeLimit);
+      _log.info('Connecting to redis');
+      final u = Uri.parse(_connectionString);
+      final socket = await Socket.connect(
+        u.host,
+        u.port,
+      ).timeout(_connectTimeLimit);
+
       // Create context
-      final client = Client(connection);
       return _RedisContext(
-        connection: connection,
-        client: client,
-        commands: client.asCommands(),
+        client: RespClient(socket, socket),
       );
+    } on RedisConnectionException {
+      throw IntermittentCacheException('connection failed');
     } on TimeoutException {
       throw IntermittentCacheException('connect failed with timeout');
-    } on SocketException {
+    } on IOException {
       throw IntermittentCacheException('connect failed with socket exception');
```

Comment:
we should probably also add the exception message here.

Reply:
Adding... `$e`, but in general, we just ignore these exceptions... they are signals to try-again :D